codegen/transforms/svc_types.sysl

```diff
@@ -309,7 +309,7 @@ CodeGenTransform:
 
       Comment = "// Code generated by sysl DO NOT EDIT.\n"
 
-      TopLevelDecl = unused | GoStructs(app.types) | GoSetType(app.types) | RequestTypes(app.endpoints, module) | validatorDecl(app.types) |  addDecl(app.types) | lookupDecl(app.types) | GoAlias(app.alias where(HasPrefix(.key, "EXTERNAL_") == false && .key != "Empty")) | GoEmptyType(app.alias where(.key == "Empty"))
+      TopLevelDecl = unused | GoStructs(app.types where(.attrs.patterns == null)) | GoSetType(app.types where(.attrs.patterns == null)) | RequestTypes(app.endpoints, module) | validatorDecl(app.types where(.attrs.patterns == null)) |  addDecl(app.types where(.attrs.patterns == null)) | lookupDecl(app.types where(.attrs.patterns == null)) | GoAlias(app.alias where(HasPrefix(.key, "EXTERNAL_") == false && .key != "Empty")) | GoEmptyType(app.alias where(.key == "Empty"))
```

Comment:
This will generate spurious warnings, should use `where("patterns" !in .attrs))` instead of `where(.attrs.patterns == null)`.

!in is missing in Sysl. I have the necessary changes: https://github.com/anz-bank/sysl/tree/add-not-in-binexpr but I need to write the tests.

Reply:
This is not merged yet, can I leave if for now and create an issue to change it after the sysl update?

Reply:
OK, but you will get complaints about the warnings.

Reply:
@gkanz Can we merge this in with a `TODO` and revisit after your changes?